File: Number_Guessing_Games.kt
import kotlin.random.Random

fun main() {
    val secretNumber = Random.nextInt(1, 101)
    var attempts = 0

    println("Bem-vindo ao jogo de adivinhação!")

    while (true) {
        print("Digite um número: ")
        val guess = readLine()?.toIntOrNull() ?: continue
        attempts++

        when {
            guess == secretNumber -> {
                println("Parabéns! Você acertou em $attempts tentativas.")
                break
            }
            guess < secretNumber -> println("Tente um número maior.")
            else -> println("Tente um número menor.")
        }
    }
}

File: Number_Guessing_Games.rs
use std::io;

fn main() {
    let secret_number = rand::random::<u32>() % 100 + 1;
    let mut attempts = 0;

    println!("Bem-vindo ao jogo de adivinhação!");

    loop {
        let mut guess = String::new();

        println!("Digite um número: ");
        io::stdin().read_line(&mut guess).expect("Falha ao ler a linha");
        let guess: u32 = match guess.trim().parse() {
            Ok(num) => num,
            Err(_) => continue,
        };

        attempts += 1;

        match guess.cmp(&secret_number) {
            std::cmp::Ordering::Equal => {
                println!("Parabéns! Você acertou em {} tentativas.", attempts);
                break;
            }
            std::cmp::Ordering::Less => println!("Tente um número maior."),
            std::cmp::Ordering::Greater => println!("Tente um número menor."),
        }
    }
}
